metadata-ingestion/src/datahub/ingestion/source/sql/vertica.py

```diff
@@ -614,13 +740,70 @@ def loop_models(
                     self.report.report_dropped(dataset_name)
                     continue
                 try:
-                    yield from self._process_models(
+                    columns: List[Dict[Any, Any]] = []
+                    dataset_urn = make_dataset_urn_with_platform_instance(
+                        self.platform,
+                        dataset_name,
+                        self.config.platform_instance,
+                        self.config.env,
+                    )
+                    dataset_snapshot = DatasetSnapshot(
+                        urn=dataset_urn,
+                        aspects=[StatusClass(removed=False)],
+                    )
+                    description, properties, location = self.get_model_properties(
+                        inspector, schema, models
+                    )
+
+                    dataset_properties = DatasetPropertiesClass(
+                        name=models,
+                        description=description,
+                        customProperties=properties,
+                    )
+
+                    dataset_snapshot.aspects.append(dataset_properties)
+                    pk_constraints: dict = {}
+                    foreign_keys: Optional[List[ForeignKeyConstraintClass]] = None
+                    schema_fields = self.get_schema_fields(dataset_name, columns)
+
+                    schema_metadata = get_schema_metadata(
+                        self.report,
                         dataset_name,
-                        inspector,
-                        schema,
-                        models,
-                        sql_config,
+                        self.platform,
+                        columns,
+                        pk_constraints,
+                        foreign_keys,
+                        schema_fields,
+                    )
+
+                    dataset_snapshot.aspects.append(schema_metadata)
+                    db_name = self.get_db_name(inspector)
+
+                    yield from self.add_table_to_schema_container(
+                        dataset_urn, db_name, schema
+                    )
+                    mce = MetadataChangeEvent(proposedSnapshot=dataset_snapshot)
+                    yield SqlWorkUnit(id=dataset_name, mce=mce)
+                    dpi_aspect = self.get_dataplatform_instance_aspect(
+                        dataset_urn=dataset_urn
                     )
+                    if dpi_aspect:
+                        yield dpi_aspect
+                    yield MetadataChangeProposalWrapper(
+                        entityType="dataset",
```

Comment:
only entityUrn and aspect args are required to `MetadataChangeProposalWrapper` - please remove the other ones

Reply:
also let's keep this logic in _process_model?